meta-resin-common/recipes-kernel/linux/kernel-devsrc.bbappend

```diff
@@ -0,0 +1,21 @@
+do_install_append() {
+   tar -czf ${B}/kernel_source.tar.gz -C "$kerneldir/../" .
+}
+
+inherit deploy
+addtask do_deploy before do_package after do_install
+
+do_deploy() {
+    cp ${B}/kernel_source.tar.gz ${DEPLOYDIR}/
+    rm ${B}/kernel_source.tar.gz
+}
+
+# Quite a few devices have random precompiled elf binaries in their
+# kernel src git trees. Remove various checks from this package to prevent
+# QA errors
+INSANE_SKIP_${PN} = "arch debug-files"
+
+# kernel-modules-headers recipe does some work on the kernel tree.
+# We'd like to make sure that we dont tarball at the same time as that
+# recipe is working on the tree
+do_install[depends] += "kernel-modules-headers:do_deploy"
```

Comment:
I'm not sure I understand. The deploy task only works on B.

Reply:
Perhaps it should be `kernel-modules-headers:do_install`
But definitely don't want  two recipes working on the source tree. `gen_mod_headers` does some magics in the kernel source tree.

Reply:
One day we will deprecate that... But in anyway wouldn't a DEPENDS on kernel-modules-headers do it?

Reply:
That is much simpler. Wonder why I was trying to add a complex task based dependency..